src/SME.SR.Application/Commands/ControleGrade/GerarRelatorioControleGradeAnalitico/GerarRelatorioControleGradeAnaliticoCommandHandler.cs

```diff
@@ -142,10 +143,90 @@ private async Task<ComponenteCurricularControleGradeDto> MapearParaComponenteDto
+            var quantidadeGrade = await mediator.Send(new ObterQuantidadeDeAulaGradeQuery(turmaId, componenteCurricularId));
+            var eventosCadastrados = await mediator.Send(new ObterEventosPorTipoCalendarioIdEPeriodoInicioEFimQuery(tipoCalendarioId, dataInicio, dataFim));
+
+            for (var data = dataInicio.AddDays(1); data <= dataFim; data = data.AddDays(1))
+            {
+                if (VerificaDataEhSegunda(data))
+                {
+                    var dataFimSemana = data.AddDays(6) > dataFim ? dataFim : data.AddDays(6);
+                    var aulasCriadas = await mediator.Send(new ObterDiasAulaCriadasPeriodoInicioEFimQuery(turmaId, componenteCurricularId, data, dataFimSemana));
+                    var diasLetivos = ObtemDiasLetivos(data, dataFimSemana, eventosCadastrados);
+
+                    visaoSemanal.Add(new VisaoSemanalControleGradeSinteticoDto()
+                    {
+                        Data = data.ToString(),
+                        QuantidadeGrade = quantidadeGrade,
+                        DiasLetivo = diasLetivos,
+                        AulasCriadas = aulasCriadas,
+                        Diferenca = PossuiDiferencaDias(quantidadeGrade,diasLetivos,aulasCriadas)
+                    });
+                }
+            }
+            return visaoSemanal;
+        }
+
+        private string PossuiDiferencaDias(int quantidadeGrade, int diasLetivos, int aulasCriadas)
+        {
+            if(quantidadeGrade > 0)
+            {
+                if (quantidadeGrade != diasLetivos)
+                    return "Sim";
+
+                if (quantidadeGrade != aulasCriadas)
+                    return "Sim";
+
+                if (diasLetivos != aulasCriadas)
+                    return "Sim";
+
+                return "Não";
+            }
+            return "Sim";
+        }
+
+        private int ObtemDiasLetivos(DateTime dataInicio, DateTime dataFim, IEnumerable<Evento> eventosCadastrados)
+        {
+            DateTimeFormatInfo dfi = DateTimeFormatInfo.CurrentInfo;
+            Calendar cal = dfi.Calendar;
+            int diasLetivos = 0;
+            for (var data = dataInicio; data <= dataFim; data = data.AddDays(1))
+            {
+                if (cal.GetDayOfWeek(data) == DayOfWeek.Saturday || cal.GetDayOfWeek(data) == DayOfWeek.Monday)
+                {
+                    if (eventosCadastrados.FirstOrDefault(a => a.DataInicio == data && a.Letivo == EventoLetivo.Sim) != null)
+                    {
+                        diasLetivos += 1;
+                    }
+                }
+                else
+                {
+                    diasLetivos += 1;
+                    if (eventosCadastrados.FirstOrDefault(a => a.DataInicio == data && a.Letivo == EventoLetivo.Nao) != null)
+                    {
+                        diasLetivos -= 1;
+                    }
+                }
+            }

```

Comment:
Não precisa usar a classe Calendar o proprio DateTime já tem a função de checar o dia da semana;
Na checagem de final de semana ta verificando se é Sabado ou Segunda;

```suggestion
            int diasLetivos = 0;
            for (var data = dataInicio; data <= dataFim; data = data.AddDays(1))
            {
                if (data.DayOfWeek == DayOfWeek.Saturday || data.DayOfWeek == DayOfWeek.Sunday)
                {
                    if (eventosCadastrados.FirstOrDefault(a => a.DataInicio == data && a.EhEventoLetivo()) != null)
                        diasLetivos += 1;
                }
                else
                {
                    diasLetivos += 1;
                    if (eventosCadastrados.FirstOrDefault(a => a.DataInicio == data && a.EhEventoNaoLetivo()) != null)
                        diasLetivos -= 1;
                }
            }

```